File: Webhook/components.js
const fuzzySet = require('fuzzyset')
const axios = require('axios');
let qanaryComponents = fuzzySet(); 

async function getComponents(){
      const response = await axios.get('http://demos.swe.htwk-leipzig.de:40111/components')
      const body = response.data 
      const newQanaryComponents = fuzzySet() 
      for (var i = 0; i < body.length; i++){   
        newQanaryComponents.add(JSON.stringify(body[i]['name']))
        } 
      qanaryComponents = newQanaryComponents  
  }
 
 

module.exports.getQanaryComponents = async () => {
  if(!qanaryComponents) {
    await getComponents()
  } 
  await getComponents() 
  return qanaryComponents
}

  
module.exports.updateComponents = function() {
  setInterval(async function(){
    await getComponents() 
    //console.log("15 sec done") 
    }, 15000);
}

File: Webhook/components.py
import urllib.request
import ast
import fuzzyset
import os
from dotenv import load_dotenv

load_dotenv()
qanaryComponents = None
qanaryComponentNames = None


def getComponents():
    global qanaryComponents
    global qanaryComponentNames
    response = urllib.request.urlopen(os.getenv('COMPONENT_URL')).read().decode()
    body = ast.literal_eval(response)
    qanaryComponents = fuzzyset.FuzzySet()
    qanaryComponentNames = set()
    for i in range(len(body)):
        qanaryComponents.add(body[i]['name'])
        # TODO
        qanaryComponentNames.add(body[i]['name'])


def getQanaryComponentNames():
    global qanaryComponentNames
    getComponents()
    return qanaryComponentNames


def getQanaryComponents():
    global qanaryComponents
    getComponents()
    return qanaryComponents

# async def updateComponents():
#     try:
#         await asyncio.wait_for(getComponents, timeout=15)
#     except:
#         print("15 sec done")
